File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        List<Integer> result = new ArrayList<>();

        for (int[] query : queries) {
            int s = query[0];
            int e = query[1];
            int k = query[2];
            int num = Integer.MAX_VALUE; // 초기값을 매우 큰 값으로 설정

            for (int i = s; i <= e; i++) {
                if (arr[i] > k) {
                    num = Math.min(num, arr[i]);
                }
            }

            if (num != Integer.MAX_VALUE) {
                result.add(num);
            } else {
                result.add(-1);
            }
        }

        // List를 배열로 변환
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/0/180000/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for (s, e, k) in queries:
        num = max(arr) + 1
        for i in range(s, e + 1):
            if arr[i] > k:
                num = min(num, arr[i])
        answer.append(num if num != max(arr) + 1 else -1)
    return answer